### ISSUE TITLE
[Console] help and merging of application arguments

### ISSUE THREAD

**Issue by bamarni** [symfony/symfony#17804]
Lately I've seen a few inconsistencies with the help command, mainly due to arguments merging. I'll use composer as an example.

```
> composer help list
Usage:
  list [options] [--] [<namespace>]

Arguments:
  namespace            The namespace name

Options:
      # Here there are only 3 options, it misses all global options
      --xml            To output list as XML
      --raw            To output raw command list
      --format=FORMAT  The output format (txt, xml, json, or md) [default: "txt"]
```

This issue is due to #3569, I think it should be reverted since @Seldaek fixed it in a more generic way here #7650.

For the help command, the extra arg is still here though :

```
> composer help
Usage:
  help [options] [--] [<command_name>]

Arguments:
  # Here it displays an extra argument at the beginning (which belongs to application)
  command                        The command to execute 
  command_name                   The command name [default: "help"]
```

I had submitted #9286 on top of @Seldaek fixes, should I rebase and check if it's still working? Or do you think it can be solved in a better way?

**Comment by fabpot** [comment]
@bamarni Please, let's rebase and see if your patch still fixes the issue.

**Comment by bamarni** [comment]
@fabpot : sure, I've rebased against 2.7 and reopened it here #19181. At the moment it's only about the second issue though (extra arg), for the first issue I think we can just revert #3569, should I give it a shot in the same PR?

**Comment by ro0NL** [comment]
@fabpot @bamarni i can confirm `bin/console help` shows the extra `command` arg on master, i covered it at https://github.com/symfony/symfony/pull/20054/files#diff-ade9d0d9f88907d4019568e74b7a35c3R4

**Comment by bamarni** [comment]
@ro0NL : thanks, so I've closed #19181 :)

**Comment by ro0NL** [comment]
Note #20054 is not merged or so.. ive closed it due lack of feedback :) I believe the bug is still there.

**Comment by ro0NL** [comment]
@fabpot do we aim to fix this? i've reopened #20054 per @l-vo's feedback. Can be rebased on 3.4
